### ISSUE TITLE
[Enhancement] Increase limit for sending to iPhones (200MB) by rewriting FileChunker / FileDigester 

### ISSUE THREAD

**Issue by schlagmichdoch** [schlagmichdoch/PairDrop#294]
This issue was deleted by GitHub (via hubot) as part of [mistakenly marking this account as spam on 17th February 2024](https://github.com/schlagmichdoch/PairDrop/issues/284). The creation date is unclear. There are also links to the deleted issue that are not updated to this issue. I decided to manually recreate it anyway in order to complete the information the repo holds even though the restored information might be outdated:

Issue by @schlagmichdoch:
---

> Also another thing I wanted to point out. When using https://sharedrop.io (which is another website with a GitHub very similar to PairDrop) between an iOS & Android device, there is no 200MB limitation. With PairDrop is there a way to remove this like with ShareDrop? I tested it on this website and it worked fine with a file over 200MB. Both use WebRTC right? So would this be possible?

_Originally posted by @Cohenl19 in https://github.com/schlagmichdoch/PairDrop/issues/210#issuecomment-1831516611_

See sharedrop.io and drop.lol

**Comment by schlagmichdoch** [comment]
The following comments were deleted by GitHub (via hubot) as part of [mistakenly marking this account as spam on 17th February 2024](https://github.com/schlagmichdoch/PairDrop/issues/284). The correct thread order and the creation date is unclear. I decided to manually restore them anyway in order to complete the information this issue holds even though the restored information might be outdated:

Comment by @schlagmichdoch:
---

Also, do not just limit it by checking the OS as the memory limit is probably different on different devices (iPhone 8 has 2 GB of RAM - iPhone 15 has 8 GB of RAM). Is it possible to check the available memory dynamically?

**Comment by schlagmichdoch** [comment]
@Cohenl19 @jayb-g I have an instance ready for testing that resolves this and increases transfer speed by quite a lot alongside many other things. Would you be open to testing it?

**Comment by jayb-g** [comment]
@schlagmichdoch sure

**Comment by schlagmichdoch** [comment]
Great, thanks!

Instance: https://pairdrop-dev.onrender.com/
You can also install the PWA by opening the share menu and selecting "Add to Home".

Known issue:
When receiving a file on mobile, the filename is incorrect when using the share menu.

**Comment by jayb-g** [comment]
@schlagmichdoch its still painstakingly slow. For a 198mb file it was showing 22mins time estimation. Did try same transfer using localsend and it transferred it in around 30secs showing 5MB/s speed

**Comment by jayb-g** [comment]
Tried it directlt from browser. 

Trying again after adding it in URL in the app.

**Comment by schlagmichdoch** [comment]
Thanks for testing!

Please do not use the Android app but rather the browser itself, as they inject their own file digester to save files directly on disk. I'll approach them after release. Also, disabling power savers might increase speed.

Sending a 1.5GB File from Ubuntu Firefox to iOS Safari on the same network got to 5-15 MB/s for me.

Which browsers and OSs are you using.

**Comment by schlagmichdoch** [comment]
Also, as this issue is about size limit, could you try to send something bigger than 200MB *to* an iOS device to confirm everything is working for bigger files?

**Comment by jayb-g** [comment]
@schlagmichdoch got it. Forgot to mention. I tested on Android chromium. Will test on Desktop browsers too

**Comment by dgpmaluco** [comment]
Hello there!

Wanted to ask if there is a fix for this?

**Comment by jayb-g** [comment]
@dgpmaluco yes it seems to be fixed on web. Not sure but may be not via app.

You can test using browser as mentioned above and report back.

**Comment by dgpmaluco** [comment]
@jayb-g I've self hosted it, version v1.11.2 and it still reports the 200 mb limit using the browser on mobile

Brave Android trying to send to iOS Safari

**Comment by mahengrui1** [comment]
Well, I don't think the web fixed it
